### ISSUE TITLE
As a user/administrator of the platform I'd like the option of collecting all logs of all products

### ISSUE THREAD

**Issue by lfrancke** [stackabletech/issues#180]
I'd like to have a solution set up where all logs from the products we deploy (not operators, there is a [separate epic](https://github.com/stackabletech/issues/issues/8) for operators) are collected and ingested into my system of choice (see below for details).

This is a research only ticket and it's done when a technology has been chosen that allows us to collect, do simple parsing (e.g. timestamps) and forward logs to downstream products.
These downstream products (e.g. a storage solution) are not in scope of this ticket.

This is done when we have chosen a tool and found a way to integrate it into our CRDs/Operators that allows me to configure what should happen with my logs and allows me to at least write to tier 1 targets and ideally also tier 2 (see below).

This tool (or combination of tools) should be able to write to (tier 1) at least Elasticsearch, OpenSearch and S3.
Three optional (tier 2) level sinks are the native logging solutions for GCP, AWS and Azure. I don't know their names except Cloudwatch for AWS but I'm sure the others have something similar.
And a tier 3 optional sink is HDFS
I'm not 100% sure how the "user interface" in our CRDs should look like but as an administrator I'd basically like to say "Move log file A from Kafka to my Elasticsearch instance and move log file B from Kafka to my Cloudwatch account"

Please prepare a decision and either call a meeting or present it at the architecture meeting.

**Comment by lfrancke** [comment]
Good question. I updated the description, this is _not_ for the operators only the products we deploy.

**Comment by teozkr** [comment]
I think something like Vector is the only reasonable option we have for getting some kind of common integration going between all products and sinks that we want to support.

Vector in particular seems like a decent fit, it is quite flexible in terms of both input and output formats, and supports all of your proposed tier 1 (OpenSearch and S3) and tier 2 (hyperscaler native) sinks. HDFS is not supported, but we may be able to patch this in if the need arises.

I tried spiking an integration between ZooKeeper and OpenSearch (https://github.com/stackabletech/zookeeper-operator/pull/450), and I'd say I ended up with an acceptable result (although there is more work to be done in terms of cleaning up the data and enriching with K8s metadata):

![image](https://user-images.githubusercontent.com/649832/159015209-007c5852-74cb-41a3-a097-0caf8f7f4587.png)

One question that remains if we do end up going with Vector is whether each operator should have support for all sinks natively (probably provided by the framework), or whether we want to deploy a separate Vector Operator that aggregates the output from all services. Currently I'm leaning towards providing this in the framework, in order to avoid the central aggregator becoming a single point of failure, but I'm open to other ideas here.
